test_plans/guppy/test_plan.md

```diff
@@ -0,0 +1,100 @@
+Guppy is a server that supports GraphQL queries on data from elasticsearch. 
+It allows you to query raw data with offset, maximum number of rows, sorting, and filters.
+Guppy is like a lightweight version of arranger, but with user-friendly GraphQL query syntax and more fine-grained access control. 
+The Windmill data explorer uses these tools to build charts, tables, and download/export functionalities.
+
+Note that the mock responses in the test_data/test_response_*.json may contain their keys in a different order than the
+actual results from the API (perform a key-by-key check in an implementation rather than a direct text comparison).
+
+[Guppy Doc](https://github.com/uc-cdis/guppy/blob/master/doc/queries.md)
+
+Guppy Endpoints: /graphql, /download
+
+Background Outline:
+	Given a data commons with a Fence deployment running at /user
+	And a a Guppy deployment running at /guppy 	
+	And a user mingfei@example.com registered in the Fence database
+	And a manifest dictionary_url value of https://s3.amazonaws.com/dictionary-artifacts/gtexdictionary/master/schema.json
+	And a test database containing case documents in '<data_file>'
+	And an ETL job completed so as to enable the use of flat queries
+	Examples:
+      (#1): |test_data/test_data_1.json|
+      (#2): |test_data/test_data_2.json|
+
+Feature: A user will be denied access to Guppy in the case that their auth is expired, and will be informed that their auth is invalid.
+	Background: (#1)
+
+	Scenario: Mingfei wants to make a query to Guppy, but his access token is expired.
+		Given mingfei@example.com's access token is expired
+		When Mingfei makes an API request to /guppy/graphql with test_query_1.json
+		Then the response will fail with a status code 401
+
```

Comment:
@ZakirG you might want to use first-person pronouns in the scenarios.

Reply:
done!